src/sys/msg_stack.cxx

```diff
@@ -41,7 +41,10 @@ int MsgStack::push(std::string message) {
 
     position++;
   };
-  return position - 1;
+  int result;
+  BOUT_OMP(critical(MsgStack))
+  result = position - 1;
+  return result;
```

Comment:
Would something like the `ScopedLock` presented [here](https://bisqwit.iki.fi/story/howto/openmp/#Locks) be useful for bits like this?

Reply:
Couldn't this be a data race if we get a bunch of threads queued at the first critical, they all enter before the first gets into the second critical section? I guess it depends on how we interpret the returned value, but I can imagine each time we call this the value of result on the different threads could be essentially random.

Reply:
Would this all be better as `BOUT_OMP(single)` so only a single thread deals with adding things to the stack?